tutorials/packaging/snap-a-website.md

```diff
@@ -0,0 +1,681 @@
+
+Electron apps show a menu by default. We can either keep it intact, customize it or hide it.
+
+Since our HTML5 game doesn't need any menu, we are going to hide it with the `setMenuBarVisibility` function:
+
+```js
+window.setMenuBarVisibility(false);
+```
+
+Let's add this line right above the `loadURL` function call, so our file looks like this:
+
+```js
+const electron = require('electron');
+const { shell, app, BrowserWindow } = electron;
+const HOMEPAGE = 'http://castlearena.io'
+
+let mainWindow;
+
+app.on('ready', () => {
+    window = new BrowserWindow({
+        width: 1200,
+        height: 900,
+        webPreferences: {
+          nodeIntegration: false
+        }
+    });
+    window.setMenuBarVisibility(false);
+    window.loadURL(HOMEPAGE);
+
+    window.on('closed', () => {
+        window = null;
+    });
+});
+
+```
+
+#### Opening external URLs in the default browser
+
+External URLs are URLs leading outside our app. Since our app is not a web browser with navigation (back or forward) buttons, let's deal with them by opening them in the default web browser.
+
+When a URL is about to be loaded, Electron triggers a `will-navigate` event. We are going to listen to these events and compare the URL they provide with the domain of our app. If they don't match, we will open the URL in the default browser.
+
+This snippet takes care of it:
+
+```js
+window.webContents.on('will-navigate', (ev, url) => {
+  parts = url.split("/");
+  if (parts[0] + "//" + parts[2] != HOMEPAGE){
+    ev.preventDefault()
+    shell.openExternal(url)
+  };
+})
+```
+
+We are going to add it to our code, so it looks like this:
+
+```js
+const electron = require('electron');
+const { shell, app, BrowserWindow } = electron;
+const HOMEPAGE = 'http://castlearena.io'
+
+let mainWindow;
+
+app.on('ready', () => {
+    window = new BrowserWindow({
+        width: 1200,
+        height: 900,
+        webPreferences: {
+          nodeIntegration: false
+        }
+    });
+    window.setMenuBarVisibility(false);
+    window.loadURL(HOMEPAGE);
+
+    window.webContents.on('will-navigate', (ev, url) => {
+      parts = url.split("/");
+      if (parts[0] + "//" + parts[2] != HOMEPAGE){
+        ev.preventDefault()
+        shell.openExternal(url)
+      };
+    });
+
+    window.on('closed', () => {
+        window = null;
+    });
+});
+```
+
+We are done with refinements, and we are almost done with the app. You can test it some more to see if it behaves as you would expect:
+
+```bash
+./node_modules/.bin/electron main.js
+```
+
+ Next, we are going to add some metadata to the app project and use electron-builder to generate an executable file.
+
+## Electron metadata (package.json)
+Duration: 2:00
+
+Now that we have a working app, we need to turn it into an executable file. To do this, we are going to use a dependency we installed earlier: electron-builder. To build the executable, it requires a `package.json` file containing some basic metadata.
```

Comment:
This is a little slow to read. What about:
`Now that we have a working app, we are going to turn it into an executable file using a dependency we installed earlier: electron-builder`.